crates/bevy_math/src/lerp.rs

```diff
@@ -0,0 +1,56 @@
+/// Provides a trait to implement the linear interpolation function.
+pub trait Lerp {
+    /// The Scalar type used in the linear interpolation function.
+    type Scalar;
+
+    /// Performs a linear interpolation between `self` and `rhs` based on the value `s`.
+    ///
+    /// When `s` is `0.0`, the result will be equal to `self`.  When `s` is `1.0`, the result
+    /// will be equal to `rhs`. When `s` is outside of range `[0, 1]`, the result is linearly
+    /// extrapolated.
+    fn lerp(self, rhs: Self, s: Self::Scalar) -> Self;
```

Comment:
Why is `Scalar` an associated type? Also, why is `s` not `Self`?

Reply:
> Why is `Scalar` an associated type? 

The user can choose to interpolate with an `f64` for instance

> Also, why is `s` not `Self`?

For complex types like `Color`